### ISSUE TITLE
C++11 is forced in gazebo-config.cmake.in

### ISSUE THREAD

**Issue by eaglesemanation** [osrf/gazebo#3143]
I'm trying to build `gazebo_ros_pkgs` on Fedora 35 directly from source. As a dependency for that I've installed `gazebo-devel` package version 10.1.0. The first error I get is this:
```
...

[4/56] Building CXX object test/CMakeFiles/test_conversions.dir/test_conversions.cpp.o
FAILED: mtest/CMakeFiles/test_conversions.dir/test_conversions.cpp.o 
/usr/lib64/ccache/c++ -DDEFAULT_RMW_IMPLEMENTATION=rmw_cyclonedds_cpp -DLIBBULLET_VERSION=3.08 -DLIBBULLET_VERSION_GT_282 -DRCUTILS_ENABLE_FAULT_INJECTION 

<A lot of -I and -isystem>

-Wall -Wextra -Wpedantic -Wno-inconsistent-missing-override -std=c++11 -I/usr/include/uuid 
-MD -MT test/CMakeFiles/test_conversions.dir/test_conversions.cpp.o 
-MF test/CMakeFiles/test_conversions.dir/test_conversions.cpp.o.d 
-o test/CMakeFiles/test_conversions.dir/test_conversions.cpp.o
-c /home/<redacted>/gazebo_ros_pkgs/src/gazebo_ros_pkgs/gazebo_ros/test/test_conversions.cpp
In file included from /home/<redacted>/ros2_galactic/install/builtin_interfaces/include/builtin_interfaces/msg/detail/time__struct.hpp:8,
                 from /home/<redacted>/ros2_galactic/install/builtin_interfaces/include/builtin_interfaces/msg/time.hpp:7,
                 from /home/<redacted>/gazebo_ros_pkgs/src/gazebo_ros_pkgs/gazebo_ros/include/gazebo_ros/conversions/builtin_interfaces.hpp:20,
                 from /home/<redacted>/gazebo_ros_pkgs/src/gazebo_ros_pkgs/gazebo_ros/test/test_conversions.cpp:17:
/home/<redacted>/ros2_galactic/install/rosidl_runtime_cpp/include/rosidl_runtime_cpp/bounded_vector.hpp:477:3: error: ‘emplace_back’ function uses ‘auto’ type specifier without trailing return type
  477 |   auto
      |   ^~~~
/home/<redacted>/ros2_galactic/install/rosidl_runtime_cpp/include/rosidl_runtime_cpp/bounded_vector.hpp:477:3: note: deduced return type only available with ‘-std=c++14’ or ‘-std=gnu++14’

...

```

gazebo_ros has configured global [CMAKE_CXX_STANDARD to be 14](https://github.com/ros-simulation/gazebo_ros_pkgs/blob/205755685f77300a27d16c9739b1e397e30811d9/gazebo_ros/CMakeLists.txt#L6), but because of [gazebo-config.cmake](https://github.com/osrf/gazebo/blob/5dffe8c8cb7a410f63a638612c2aded71d3bdb72/cmake/gazebo-config.cmake.in#L139) it's being overridden. When I manually remove this line from `/usr/lib64/cmake/gazebo/gazebo-config.cmake` all errors related to C++14 disappear.

**Comment by alexdewar** [comment]
I wonder if setting the C++ standard could be entirely omitted. How many compilers don't target C++11 (or newer) by default nowadays anyway?

**Comment by traversaro** [comment]
> gazebo_ros has configured global [CMAKE_CXX_STANDARD to be 14](https://github.com/ros-simulation/gazebo_ros_pkgs/blob/205755685f77300a27d16c9739b1e397e30811d9/gazebo_ros/CMakeLists.txt#L6), but because of [gazebo-config.cmake](https://github.com/osrf/gazebo/blob/5dffe8c8cb7a410f63a638612c2aded71d3bdb72/cmake/gazebo-config.cmake.in#L139) it's being overridden.

That code was removed in https://github.com/gazebosim/gazebo-classic/pull/3050/files and released in  11.10.0, so I think we can close the issue. Feel free to re-open if there are still problems.
